User can create an entry

t-doan opened the pull request:
![user-can-create-an-entry](https://github.com/t-doan/code-journal/assets/26643555/67257b64-9b22-4e61-b94d-44d9e3e2eb11)

t-doan committed:
Create the remaining HTML for the header in the header-container #1

t-doan committed:
Create the html form. Complete the form using responsive layout and able to adjust. #1

t-doan committed:
make all inputs required.

t-doan committed:
reload eslint, added event listener for input image urls #1

t-doan committed:
Add submit event listener. Prevents page from refreshing. Store inputs into new objet. Reset form once submitted. #1

t-doan committed:
added a 'beforeunload' listener  to window object. Serialize the data model as JSON and store it to localStorage. Have a conditional to check for data models stored in the localStorage. If any, parse and assign it to the data variable so user can continue adding entries. App has been tested to save data models after page reloads. New journal entry will continue the entryId property #1

t-doan committed:
cleaning code

shawnkost commented on index.html:
```diff
@@ -16,12 +16,50 @@
-      <div data-view="entry-form"></div>
-      <div data-view="entries" class="hidden"></div>
+      <form id="entry-form">
+        <div data-view="entry-form">
```
These should be swapped. The `form` should be inside the `div`

shawnkost commented on js/main.js:
```diff
@@ -0,0 +1,30 @@
+/* global data */
+
+// console.log(data);
```
Remove all old console logs

shawnkost commented on js/main.js:
```diff
@@ -0,0 +1,30 @@
+  };
+
+  data.nextEntryId++;
+  data.entries.push(entryValues);
```
You want to add this to the beginning of the array rather than the end

shawnkost reviewed (changes requested):
Looking good so far Tai! A few changes here

t-doan committed:
form tag is inside the div tag.
removed old console.log in main.js
change push method to unshift method so that entries will be the first index of the array #1

t-doan commented on index.html:
```diff
@@ -22,8 +22,8 @@ <h1 class="margin-none header-padding">Code Journal</h1>
-      <form id="entry-form">
-        <div data-view="entry-form">
+      <div data-view="entry-form">
+        <form id="entry-form">
```
made the changes

t-doan commented on index.html:
```diff
@@ -57,9 +57,9 @@ <h1>New Entry</h1>
-      </form>
+        </form>
+      </div>
+      <div data-view="entries" class="hidden"></div>
```
made the changes

t-doan commented on js/main.js:
```diff
@@ -1,7 +1,5 @@
 /* global data */
 
-// console.log(data);
-
```
removed the console.log

t-doan commented on js/main.js:
```diff
@@ -24,7 +22,7 @@ function handleSubmit(event) {
 
   data.nextEntryId++;
-  data.entries.push(entryValues);
+  data.entries.unshift(entryValues);
```
changed to `unshift` method instead of `push`

t-doan replied:
New entries will be the first index of the array

![image](https://github.com/t-doan/code-journal/assets/26643555/31ba97a4-7163-42b9-ac53-31de0f341dfa)

shawnkost reviewed (approved):
👍🏻
